src/modules/mavlink/mavlink_main.cpp

```diff
@@ -2397,7 +2397,12 @@ Mavlink::task_main(int argc, char *argv[])
 							_vehicle_command_ack_sub.get_last_generation());
 					}
 
-					if (!command_ack.from_external && command_ack.command < vehicle_command_s::VEHICLE_CMD_PX4_INTERNAL_START) {
+					const bool is_target_known = _receiver.component_was_seen(command_ack.target_system, command_ack.target_component);
```

Comment:
Calling into the receiver with no locking feels kind of sketchy, but I suppose it's technically fine given the fixed component state structures and we only care about it being "seen" (and it's never marked unseen).

Reply:
Hm, so the receiver is on a different thread? In that case we should probably lock somewhere.

Reply:
Yes...

Reply:
There are a few other calls already going back and forth between the RX and TX threads, let's do a quick review overall.